classes/submission/genre/Genre.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace PKP\submission\genre;
+
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Casts\Attribute;
+use Illuminate\Database\Eloquent\Relations\HasMany;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Support\Facades\DB;
+use PKP\submissionFile\SubmissionFile;
+
+
+
+class Genre extends Model
+{
+    const GENRE_CATEGORY_DOCUMENT = 1;
+    const GENRE_CATEGORY_ARTWORK = 2;
+    const GENRE_CATEGORY_SUPPLEMENTARY = 3;
+    protected $table = 'genres';
+    protected $primaryKey = 'genre_id';
+    public $timestamps = false;
+
+    protected $fillable = [
+        'context_id', 'seq', 'enabled', 'category', 'dependent', 
+        'supplementary', 'required', 'entry_key'
+    ];
+
+    // Relationships
+    public function submissionFiles(): HasMany
+    {
+        return $this->hasMany(SubmissionFile::class, 'genre_id');
+    }
+
+
+    // Accessors and Mutators
+    protected function entryKey(): Attribute
+    {
+        return Attribute::make(
+            get: fn($value) => strtoupper($value),
+            set: fn($value) => strtolower($value)
+        );
+    }
+
+    protected function enabled(): Attribute
```

Comment:
If we're just looking to cast a PHP boolean into a database int...
- It might be better to use [attribute casting](https://laravel.com/docs/11.x/eloquent-mutators#attribute-casting), or
- better still, just leave it an integer and at some point we'll declare boolean columns in the DB with the [`boolean` type](https://laravel.com/docs/11.x/migrations#column-method-boolean) so that everything matches nicely without needing casting.